[Secure JSON RPC] Refactor the lightweight JSON RPC client to support optional versions.

JoshLind opened the pull request:
## Motivation

This small PR makes the 'version' field optional in the 'get_account_state()' API call of the secure JSON RPC client. This prevents the client from having to "sync up" to the latest state of the blockchain, and reflects recent changes to the JSON RPC server code (see  here: https://github.com/libra/libra/pull/3460).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

All local tests pass (including the newly added one).

## Related PRs

None.

davidiw commented on secure/json-rpc/src/lib.rs:
```diff
@@ -63,18 +63,71 @@ impl From<FromHexError> for Error {
+        version: Option<u64>,
+        ledger_version: Option<u64>,
+    ) -> Result<Response, Error> {
+        let method = "get_account_state_with_proof".to_string();
```
use into

JoshLind replied:
Done (and changed all other to_string() to into(), where appropriate). Thanks!

davidiw commented on secure/json-rpc/src/lib.rs:
```diff
@@ -63,18 +63,71 @@ impl From<FromHexError> for Error {
+    // Sends a submit() request to the JSON RPC server using the given transaction.
+    fn submit_transaction(&self, signed_transaction: SignedTransaction) -> Result<Response, Error> {
+        let method = "submit".to_string();
+        let params = vec![Value::String(hex::encode(lcs::to_bytes(
```
seems like something we could make a function for?
  to_value_string<T: Serialize>(value: T)

davidiw replied:
unless it is done only once...

JoshLind replied:
This is only done, once, so I've left it as is! Thanks :)

JoshLind commented:
Thanks, David! As discussed offline, I've updated this PR to only make the version field optional in the secure JSON RPC client and removed the changes that defined a JsonRpcClient interface. 

For external reader context: the JsonRpcClient interface is no longer needed as we'll test the (up and comign) JsonRpcLibraInterface implementation using end to end tests rather than unit tests, thus we don't need to expose the interface.

davidiw commented:
@bors-libra r+

bors-libra commented:
:pushpin: Commit f0d5cd4 has been approved by `davidiw`

<!-- @bors-libra r=davidiw f0d5cd4cd5e00142ee117433e13f35176827a9f6 -->

bors-libra commented:
:hourglass: Testing commit f0d5cd4cd5e00142ee117433e13f35176827a9f6 with merge ac4edd834089ddfb585b6e3fd858696d90f333c5...

bors-libra commented:
:sunny: Test successful - [checks-actions_land_blocking_test](https://github.com/libra/libra/runs/603162155), [checks-circle_commit_workflow](https://circleci.com/workflow-run/7ce9f792-fe08-4f58-9e6d-d4e911f1bfff?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)
Approved by: davidiw
Pushing ac4edd834089ddfb585b6e3fd858696d90f333c5 to master...
